bin/install

```diff
@@ -23,13 +23,19 @@ install_php() {
   if [[ $operating_system =~ "Darwin" ]]; then
     exit_if_homebrew_not_installed
 
+    local argon2_path=$(homebrew_package_path argon2)
     local bison_path=$(homebrew_package_path bison)
     local icu4c_path=$(homebrew_package_path icu4c)
     local krb5_path=$(homebrew_package_path krb5)
     local libedit_path=$(homebrew_package_path libedit)
     local libxml2_path=$(homebrew_package_path libxml2)
     local openssl_path=$(homebrew_package_path openssl@1.1)
 
+    # argon2 is optional
+    if [ -n "$argon2_path" ]; then
+      export "PKG_CONFIG_PATH=${argon2_path}/lib/pkgconfig:${PKG_CONFIG_PATH}"
```

Comment:
Is this necessary? I'm not sure what role `pkg-config` plays with this dependency. Is `--with-argon2=/foo` enough?

Reply:
PHP doesn't recognize `--with-argon2` as an option.

[The PHP docs say:](https://www.php.net/manual/en/password.installation.php)

> Prior to PHP 8.1.0, it was possible to specify the argon2 directory with `--with-password-argon2[=DIR]`.

So, I asked about it: https://chat.stackoverflow.com/transcript/message/56607367#56607367

```
ramsey: The password hashing docs say, "Prior to PHP 8.1.0, it was possible to specify the argon2 directory with --with-password-argon2[=DIR]." So, does that mean libargon2 is bundled with php-src now? php.net/manual/en/password.installation.php
Derick: No, you just can't specify the directory.
Derick: The "new" way depends on pkg-config and it supplying the right directory /cc @ramsey
ramsey: gotcha
Derick: (I didn't know that, I had to research it)
```

Most everything is moving away from (or has already moved away from) specifying the directory and using `pkgconfig` instead.

Reply:
The `configure` script uses `pkg-config` like the following to get the linker and C flags for a package, which will then go into the generated `Makefile` and passed along to the compiler when PHP is built.

A command like this:

``` shell
PKG_CONFIG_PATH="$(brew --prefix argon2)/lib/pkgconfig" \
    pkg-config --libs --cflags libargon2
```

Will return something like this:

```
-I/opt/homebrew/Cellar/argon2/20190702_1/include -L/opt/homebrew/Cellar/argon2/20190702_1/lib -largon2
```